### PR TITLE
Scss.forms palettes

### PR DESCRIPTION
**Pull request by jeremie-lucca**
- Form elements  (input, textarea, radio, checkbox, switch) colors in theming
- Default secondary color
- Adding palettes for each form element

image

image

image

image

image

### CODE REVIEW

File: packages/scss/src/components/_checkbox.scss [hunk 1]
```diff
@@ -62,6 +62,27 @@
...
+		.checkbox-input:checked ~ .checkbox-label {
+			&::before {
+				background-color: _get($palette, "color");
+				box-shadow: inset 0 0 0 2px _get($palette, "color");
+			}
+		}
+	}
+
+	// palette colors
+	@each $name, $palette in _palettes() {
```
Comment (lucienbertin) [inline packages/scss/src/components/_checkbox.scss]: you dont define default coloring here, it should be here that you use the code on lines 12-13

File: packages/scss/src/components/_input.scss [hunk 2]
```diff
@@ -93,6 +93,28 @@
...
+			box-shadow: 0 2px 0 0 _get($palette, "color");
+
+			~ .input-label {
+				color: _get($palette, "color");
+			}
+		}
+	}
+
+	// palette colors
+	@each $name, $palette in _palettes() {
```
Comment (lucienbertin) [inline packages/scss/src/components/_input.scss]: same here

File: packages/scss/src/components/_checkbox.scss (hunk 1 above)
Comment (lucienbertin) [inline packages/scss/src/components/_checkbox.scss]: see whats done for buttons [here](https://github.com/LuccaSA/lucca-front/blob/master/packages/scss/src/components/_button.scss#L60)

File: packages/scss/src/components/_radio.scss [hunk 3]
```diff
@@ -61,6 +61,32 @@
...
+
+		.radio-label {
+			&::after {
+				background: _get($palette, "color");
+			}
+		}
+	}
+
+	// palette colors
+	@each $name, $palette in _palettes() {
```
Comment (lucienbertin) [inline packages/scss/src/components/_radio.scss]: again

File: packages/scss/src/components/_switch.scss [hunk 4]
```diff
@@ -45,6 +45,31 @@
...
+			}
+
+			&::after {
+				background: _get($palette, "color");
+			}
+		}
+	}
+
+	// palette colors
+	@each $name, $palette in _palettes() {
```
Comment (lucienbertin) [inline packages/scss/src/components/_switch.scss]: again here

File: packages/scss/src/components/_textarea.scss [hunk 5]
```diff
@@ -39,6 +39,29 @@
...
+			box-shadow: 0 2px 0 0 _get($palette, "color");
+
+			~ .textarea-label {
+				color: _get($palette, "color");
+			}
+		}
+	}
+
+	// palette colors
+	@each $name, $palette in _palettes() {
```
Comment (lucienbertin) [inline packages/scss/src/components/_textarea.scss]: ditto

File: packages/scss/src/theming/components/_checkbox.theme.scss [hunk 6]
```diff
@@ -0,0 +1,4 @@
+$checkbox: (
+  color: _color("secondary"),
```
Comment (lucienbertin) [inline packages/scss/src/theming/components/_checkbox.theme.scss]: you just have to define `default-palette: _theme("palettes.secondary")` here and you're good

**Comment by jeremie-lucca** [comment]
.switch-input has been encapsuled in .switch because theming increase specificity.

**Pull request merged**